lib/src/value/calculation.dart

```diff
@@ -584,32 +615,54 @@ final class SassCalculation extends Value {
   static Object operate(
           CalculationOperator operator, Object left, Object right) =>
       operateInternal(operator, left, right,
-          inLegacySassFunction: false, simplify: true);
+          inLegacySassFunction: null, simplify: true, warn: null);
 
-  /// Like [operate], but with the internal-only [inLegacySassFunction] parameter.
+  /// Like [operate], but with the internal-only [inLegacySassFunction] and
+  /// [warn] parameters.
   ///
-  /// If [inLegacySassFunction] is `true`, this allows unitless numbers to be added and
-  /// subtracted with numbers with units, for backwards-compatibility with the
-  /// old global `min()` and `max()` functions.
+  /// If [inLegacySassFunction] isn't null, this allows unitless numbers to be
+  /// added and subtracted with numbers with units, for backwards-compatibility
+  /// with the old global `min()` and `max()` functions. This emits a
+  /// deprecation warning using the string as the function's name.
   ///
   /// If [simplify] is `false`, no simplification will be done.
+  ///
+  /// The [warn] callback is used to surface deprecation warnings.
+  ///
+  /// @nodoc
   @internal
   static Object operateInternal(
       CalculationOperator operator, Object left, Object right,
-      {required bool inLegacySassFunction, required bool simplify}) {
+      {required String? inLegacySassFunction,
+      required bool simplify,
+      required void Function(String message, [Deprecation? deprecation])?
+          warn}) {
     if (!simplify) return CalculationOperation._(operator, left, right);
     left = _simplify(left);
     right = _simplify(right);
 
     if (operator case CalculationOperator.plus || CalculationOperator.minus) {
-      if (left is SassNumber &&
-          right is SassNumber &&
-          (inLegacySassFunction
-              ? left.isComparableTo(right)
-              : left.hasCompatibleUnits(right))) {
-        return operator == CalculationOperator.plus
-            ? left.plus(right)
-            : left.minus(right);
+      if (left is SassNumber && right is SassNumber) {
+        var compatible = left.hasCompatibleUnits(right);
+        if (!compatible &&
+            inLegacySassFunction != null &&
+            left.isComparableTo(right)) {
+          warn!(
+              "In future versions of Sass, $inLegacySassFunction() will be "
+              "interpreted as the CSS $inLegacySassFunction() calculation. "
+              "This doesn't allow unitless numbers to be mixed with numbers "
+              "with units. If you want to use the Sass function, call "
+              "math.$inLegacySassFunction() instead.\n"
+              "\n"
+              "See https://sass-lang.com/d/import",
+              Deprecation.globalBuiltin);
+          compatible = true;
+        }
+        if (compatible) {
```

Comment:
Isn't it better to just make `compatible` a `const` and return
```
return operator == CalculationOperator.plus
              ? left.plus(right)
              : left.minus(right);
```
either way? Since `compatible` is set to `true` if it's false.

Reply:
`compatible` is only set to true if `inLegacySassFunction != null && left.isComparableTo(right)`.